airbyte-integrations/connectors/source-google-ads/unit_tests/common.py

```diff
@@ -38,9 +40,22 @@ def search(self, search_request):
         raise GoogleAdsException("This mocked class always returns an error")
 
 
+def make_google_ads_exception(failure_code: int):
```

Comment:
it was easier to create a full on exception object rather than create a deeply nested mock